include/type.hpp

```diff
@@ -74,11 +80,39 @@ class PtrType : public Type {
   }
 
   bool IsEqual(const Type& that) const noexcept override;
+  std::size_t ToSize() const override;
   std::string ToString() const override;
   std::unique_ptr<Type> Clone() const override;
 
  private:
   std::unique_ptr<Type> base_type_;
+  const std::size_t size_ = 8;
```

Comment:
Consider having the `ToSize` function return a constant `8` directly instead of having a constant data member.